app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1158,6 +1171,48 @@ public void populatePlaces(final LatLng currentLatLng,
         }
     }
 
+    /**
+     *  Reloads the Nearby map
+     *  Clears all location markers, refreshes them, reinserts them into the map.
+     *
+     */
+    private void reloadMap() {
+        clearAllMarkers(); // Clear the list of markers
+        binding.map.getController().setZoom(ZOOM_LEVEL); // Reset the zoom level
+        binding.map.getController().setCenter(lastMapFocus); // Recenter the focus
```

Comment:
Refreshing seems to move the map to the place I am in physically, is it expected?

https://github.com/user-attachments/assets/76cd2cad-4ce3-4048-91d8-83c048e3eb18

Reply:
> Refreshing seems to move the map to the place I am in physically, is it expected?
> 
>  screen-20241025-141156.mp4

My colleague @MarcusBarta worked on the map reload part of this feature, and he has told me that this is expected.

Reply:
Understood! I created https://github.com/commons-app/apps-android-commons/issues/5906 as an enhancement for this.